packages/block-library/src/site-logo/edit.js

```diff
@@ -0,0 +1,283 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { debounce } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { isBlobURL } from '@wordpress/blob';
+import { useState, useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import {
+	useEntityProp,
+	__experimentalUseEntitySaving,
+} from '@wordpress/core-data';
+import {
+	IconButton,
+	Notice,
+	PanelBody,
+	RangeControl,
+	ResizableBox,
+} from '@wordpress/components';
+import {
+	BlockControls,
+	BlockAlignmentToolbar,
+	BlockIcon,
+	InspectorControls,
+	MediaPlaceholder,
+	MediaReplaceFlow,
+} from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import icon from './icon';
+
+import ImageSize from '../image/image-size';
+
+const ALLOWED_MEDIA_TYPES = [ 'image' ];
+const ACCEPT_MEDIA_STRING = 'image/*';
+const getHandleStates = ( align, isRTL = false ) => {
+	const defaultAlign = isRTL ? 'right' : 'left';
+	const handleStates = {
+		left: {
+			right: true,
+			left: false,
+		},
+		center: {
+			right: true,
+			left: true,
+		},
+		right: {
+			right: false,
+			left: true,
+		},
+	};
+
+	return handleStates[ align ? align : defaultAlign ];
+};
+
+export default function LogoEdit( { attributes: { align, width }, children, className, clientId, setAttributes, isSelected } ) {
+	const [ isEditing, setIsEditing ] = useState( false );
+	const [ url, setUrl ] = useState( null );
+	const [ error, setError ] = useState();
+	const [ logo, setLogo ] = useEntityProp( 'root', 'site', 'sitelogo' );
+	const [ isDirty, , save ] = __experimentalUseEntitySaving(
+		'root',
+		'site',
+		'sitelogo'
+	);
```

Comment:
Not sure why the saving is needed in the block itself, should we rely on the multi-entity saving flow instead (basically, just remove the saving code)

Reply:
Removed